worldedit-bukkit/src/main/java/com/sk89q/worldedit/bukkit/BukkitPlayer.java

```diff
@@ -77,13 +77,24 @@
     private final WorldEditPlugin plugin;
     private final PermissionAttachment permAttachment;
 
-    public BukkitPlayer(Player player) {
+    /**
+     * This constructs a new {@link BukkitPlayer} for the given {@link Player}.
+     * 
+     * @param player The corresponding {@link Player} or null if they are offline
+     */
+    public BukkitPlayer(@Nullable Player player) {
         super(getExistingMap(WorldEditPlugin.getInstance(), player));
         this.plugin = WorldEditPlugin.getInstance();
         this.player = player;
         this.permAttachment = plugin.getPermissionAttachmentManager().addAttachment(player);
```

Comment:
This will shift any NPEs into the PermissionAttachmentManager.

Reply:
Ah I see, should I add another null check for the PermissionAttachmentManager? (Although the parameter being nullable feels a bit off)
Or should I add another if-statement to this assignment and make `permAttachment` nullable as a result of that?

Reply:
Within the call to super constructor, https://github.com/IntellectualSites/FastAsyncWorldEdit/blob/main/worldedit-bukkit/src/main/java/com/sk89q/worldedit/bukkit/WorldEditPlugin.java#L518 is also called ultimately, which will also NPE on the player.

It's worth noting the default constructor in AbstractPlayerActor at https://github.com/IntellectualSites/FastAsyncWorldEdit/blob/main/worldedit-core/src/main/java/com/sk89q/worldedit/extension/platform/AbstractPlayerActor.java#L86 creates an empty map, _not_ null.

Regarding the PermissionAttachmentManager, nullability is something that @aurorasmiles or @weaondara are best suited to comment on.

Reply:
I see, I just went ahead and checked WorldGuard's code again, fortunately most methods throw an `UnsupportedOperationException` anyway, as they are not meant for offline players:
https://github.com/EngineHub/WorldGuard/blob/master/worldguard-bukkit/src/main/java/com/sk89q/worldguard/bukkit/BukkitOfflinePlayer.java

So making the permission attachment nullable may not have any big implications? As this is a pretty edge case reminiscient of "mocking" the object. So under normal circumstances this shouldn't disturb the current mechanics.
I am open to any suggestions, I think making this work with WorldGuard's intended usage would be beneficial, but I realize that this may be a bit deeper than I first anticipated.

How should we proceed?
I thought about creating an OfflinePermissionAttachment, but the constructor takes a NotNull Permissible, even if that check isn't enforced right now.

Reply:
Regardless of whether or not the PermissionAttachment should "work" or not when the player is null, there are still issues with NPE during the initial object instantiation, the fixing of which is presumably the main aim of the PR. I think adding a nullcheck on the player and then simply returning null should be fine given the further information you've provided.

Reply:
Yes, that is indeed the aim of the PR.
Would you prefer the null-check to happen in the AttachmentManager or inside BukkitPlayer?

Reply:
nullchecks should usually happen just before a nullability issue

Reply:
Ignore the BukkitPermissionAttachmentManager for now, as that might be changed on main branch outside of this PR